tools/autograd/derivatives.yaml

```diff
@@ -238,14 +238,14 @@
 
 - name: addmm(Tensor self, Tensor mat1, Tensor mat2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
   self: maybe_multiply(grad, beta.conj())
-  mat1: mm_mat1_backward(grad, mat2, mat1.sizes(), mat1.strides(), mat1.layout(), alpha)
-  mat2: mm_mat2_backward(grad, mat1, mat2.sizes(), mat2.strides(), mat2.layout(), alpha)
+  mat1: mm_mat1_backward(grad, mat2, mat1.sym_sizes(), mat1.sym_strides(), mat1.layout(), alpha)
```

Comment:
Note that we do need to add rules for the sym_* versions to https://github.com/pytorch/pytorch/blob/fcb124406bdf86bc2d15e999d5a3e09b86238bba/tools/autograd/load_derivatives.py#L746 before we can do such replacements.

Reply:
To be clear, I don't intend to land any of this PR as is right now, haha. It's just collecting a bunch of hacks I needed for an E2E training of a MLP with dynamic shapes.

Reply:
Sure, just wanted to bring it up because we already have a couple of these in some formulas and we shouldn't forget to add these rules to avoid memory issues ;)